internal/identity/manager/datastore_test.go

```diff
@@ -0,0 +1,78 @@
+package manager
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/protobuf/testing/protocmp"
+
+	"github.com/pomerium/pomerium/pkg/grpc/session"
+	"github.com/pomerium/pomerium/pkg/grpc/user"
+)
+
+func TestDataStore(t *testing.T) {
+	t.Parallel()
+
+	ds := newDataStore()
+	s, u := ds.getSessionAndUser("S1")
+	assert.Nil(t, s, "should return a nil session when none exists")
+	assert.Nil(t, u, "should return a nil user when none exists")
+
+	u, ss := ds.getUserAndSessions("U1")
+	assert.Nil(t, u, "should return a nil user when none exists")
+	assert.Empty(t, ss, "should return an empty list of sessions when no user exists")
+
+	s = &session.Session{Id: "S1", UserId: "U1"}
+	ds.putSession(s)
+	s1, u1 := ds.getSessionAndUser("S1")
+	assert.NotNil(t, s1, "should return a non-nil session")
+	assert.False(t, s == s1, "should return different pointers")
```

Comment:
totally optional: you might like to use https://pkg.go.dev/github.com/stretchr/testify/assert#NotSame for these pointer assertions